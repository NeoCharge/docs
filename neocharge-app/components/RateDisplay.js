import React from "react";
import { StyleSheet, View, Text, Dimensions } from "react-native";

export default class RateDisplay extends React.Component {
    constructor(props) {
        super(props)
        this.state = {
            rawDeviceLogs: this.props.data,
        }
    }

    getTotalCharge() {
        const rawList = this.props.data
        var total = 0;
        var today = Date.parse(new Date().toISOString().split('T')[0]);
        var oneDay = 1000 * 60 * 60 * 24;

        rawList.forEach(function (obj) {
            var date = Date.parse(obj.startTime.substring(0, 10));
            var dateDiff = (today - date) / oneDay;

            if (dateDiff < 100)
                total += obj.priPower;
        }
        )

        return total;
    }

    render() {

        return (
            <View style={styles.container}>
                <Text style={styles.textNum}>{this.getTotalCharge()}</Text>
                <Text style={styles.textUnit}>kWh</Text>
            </View>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        padding: 30,
        borderRadius: 15,
        width: "40%",
        height: "20%",
        justifyContent: "center",
        alignItems: "center",
        backgroundColor: "#ffffff"
    },
    textNum: {
        fontSize: 35
    },
    textUnit: {
        fontSize: 20
    }
});