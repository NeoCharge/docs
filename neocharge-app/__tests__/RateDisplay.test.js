import React from 'react';
import { shallow } from 'enzyme';
import RateDisplay from '../RateDisplay';

describe('RateDisplay', () => {
    describe('Rendering', () => {
        it('should match to snapshot', () => {
            const component = shallow(<RateDisplay data="" />)
            expect(component).toMatchSnapshot()
        });
    });
});