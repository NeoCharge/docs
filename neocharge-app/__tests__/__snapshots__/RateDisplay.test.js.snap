// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RateDisplay Rendering should match to snapshot 1`] = `ShallowWrapper {}`;
